crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -332,6 +341,196 @@ impl<P: FsProvider> ResctrlPlugin<P> {
             self.emit_pod_add_or_update(&pod_uid, ps);
         }
     }
+
+    /// Try to create a resctrl group for a pod if currently Failed.
+    /// Emits AddOrUpdate only on state transition.
+    pub fn retry_group_creation(&self, pod_uid: &str) -> resctrl::Result<ResctrlGroupState> {
+        use std::io;
+        use std::path::PathBuf;
+
+        // Snapshot decision under lock
+        let should_try = {
+            let st = self.state.lock().unwrap();
+            match st.pods.get(pod_uid) {
+                Some(ps) => matches!(ps.group_state, ResctrlGroupState::Failed),
+                None => {
+                    return Err(resctrl::Error::Io {
+                        path: PathBuf::from("<pod>"),
+                        source: io::Error::from_raw_os_error(libc::ENOENT),
+                    })
+                }
+            }
+        };
+
+        if !should_try {
+            // Either does not exist or already Exists; return current state
+            let st = self.state.lock().unwrap();
+            let current = st.pods.get(pod_uid).unwrap().group_state.clone();
+            return Ok(current);
+        }
+
+        // Drop lock while performing filesystem operation
+        let res = self.resctrl.create_group(pod_uid);
+        match res {
+            Ok(path) => {
+                let mut st = self.state.lock().unwrap();
+                // Re-check and update if still Failed
+                if let Some(ps) = st.pods.get_mut(pod_uid) {
+                    if matches!(ps.group_state, ResctrlGroupState::Failed) {
```

Comment:
change this to a match statement that has branches for Failed an Exists. This way if we change ResctrlGroupState to add more enum values, the compiler will flag this check to accommodate the new option